src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -2477,6 +2477,55 @@ public IEnumerable<Document> GetRevisionsBinEntries(DocumentsOperationContext co
             }
         }
 
+        public IEnumerable<Document> GetRevisionsBinEntries(DocumentsOperationContext context, DateTime before, DocumentFields fields, RevisionsBinCleanerState state)
+        {
+            var table = new Table(RevisionsSchema, context.Transaction.InnerTransaction);
+
+            var ended = false;
+            while (ended == false)
+            {
+                ended = true;
+
+                using (GetEtagAsSlice(context, state.LastEtag, out var startSlice))
+                {
+                    foreach (var seekResult in table.SeekForwardFrom(RevisionsSchema.Indexes[DeleteRevisionEtagSlice], startSlice, 0))
+                    {
+                        var tvr = seekResult.Result;
+                        var storageId = tvr.Reader.Id;
+                        var etag = TableValueToEtag((int)RevisionsTable.DeletedEtag, ref tvr.Reader);
+                        if (etag == NotDeletedRevisionMarker)
+                        {
+                            context.Transaction.InnerTransaction.ForgetAbout(storageId);
+                            continue;
+                        }
+
+                        state.LastEtag = etag;
+
+                        using (TableValueToSlice(context, (int)RevisionsTable.LowerId, ref tvr.Reader, out Slice lowerId))
+                        {
+                            if (IsRevisionsBinEntry(context, table, lowerId, etag) == false) // if its not last - continue
+                            {
+                                context.Transaction.InnerTransaction.ForgetAbout(storageId);
+                                continue;
+                            }
+                        }
+
+                        var deleteRevision = TableValueToRevision(context, ref tvr.Reader, fields);
+                        if (deleteRevision.LastModified >= before)
+                        {
+                            context.Transaction.ForgetAbout(deleteRevision);
+                            yield break;
+                        }
+
+                        yield return deleteRevision;
+                        context.Transaction.InnerTransaction.ForgetAbout(storageId);
```

Comment:
is this correct? shouldn't everything that uses this method be wrapped in TransactionForgetAboutDocumentEnumerator ?

Reply:
I don't think so,
because we use here `SeekForwardFrom` which uses TreeIterator.MoveNext() and not TransactionForgetAboutDocumentEnumerator.MoveNext()..
So I think we need to handle the `ForgetAbout` by ourself here :\

Reply:
Wrapped it with TransactionForgetAboutDocumentEnumerator